src/views/Jobs/JobResultsTable.vue

```diff
@@ -914,6 +972,89 @@ export default Vue.extend({
                     }
                   });
                 });
+              } else if (
+                jobPrediction.method === "PathwayPredictor+CofactorSwap"
+              ) {
+                // ids of reactions which should be knocked out
+                // and then added with swapped cofactors
+                const knockoutIds: string[] = [];
+                const addedReactions: Object[] = [];
+                jobPrediction.manipulations.forEach(manipulation => {
+                  knockoutIds.push(manipulation.id);
+                  const reactionToAdd = Object.assign(
+                    {},
+                    this.model["model_serialized"]["reactions"].find(
+                      r => r.id === manipulation.id
+                    )
+                  );
+                  // swap cofactors
+                  for (let i = 0; i < manipulation.from.length; i++) {
+                    reactionToAdd.metabolites[manipulation.to[i]] =
+                      reactionToAdd.metabolites[manipulation.from[i]];
+                    delete reactionToAdd.metabolites[manipulation.from[i]];
+                  }
+                  // needed for reaction string
+                  const substrates: string[] = [];
+                  const products: string[] = [];
+                  // metabolites should be array of objects
+                  const metabolites = Object.keys(
+                    reactionToAdd.metabolites
+                  ).map(metaboliteId => {
+                    const stoichiometry =
+                      reactionToAdd.metabolites[metaboliteId];
+                    stoichiometry < 0
+                      ? substrates.push(
+                          Math.abs(stoichiometry) + " " + metaboliteId
+                        )
+                      : products.push(stoichiometry + " " + metaboliteId);
```

Comment:
I don't think sideeffects in ternary operator are obvious when reading the code.
And also one scope higher, sideeffects inside array.map (instead of array.forEach)

Reply:
Thanks! Changed that